include/dsn/cpp/utils.h

```diff
@@ -425,7 +424,7 @@ namespace dsn {
 
         typedef struct _tls_tid
         {
-            int magic;
+            unsigned int magic;
```

Comment:
Well, this is my bad. Coz we encountered several bugs which can be detected by the compiler warnings, so I decides to turn the compile warnings on. And I add compile options "-Wall -Werror". As a result, lots of warnings of "signed/unsigned compare" are generated. "Magic" always compared with "0xdeadbeef", which is unsigned. 
I admit the options like "-Wall -Werror" is a bit too strict. So I change options to "-Wall -Werror -Wno-sign-compare". But I strongly suggest that we should never ignore the compiler warnings.